### PR TITLE
MAUI - 886398 - Prepared the UG documentation for DataGrid column drag and drop feature.

### PR DESCRIPTION
**Pull request by Tamilarasan-Paranthaman**
## Description

- Prepared the UG documentation for DataGrid column drag and drop feature.

### CODE REVIEW

File: maui-toc.html [hunk 1]
```diff
@@ -492,6 +492,7 @@
             <li><a href="/maui/DataGrid/column-types">Column Types</a></li>
             <li><a href="/maui/DataGrid/column-sizing">Column Sizing</a></li>
             <li><a href="/maui/DataGrid/column-resizing">Column Resizing</a></li>
+            <li><a href="/maui/DataGrid/column-drag-and-drop">Column Drag and Drop</a></li>
```
Comment (ashok-kuvaraja) [inline maui-toc.html:495]: Add after pull to refresh topic.

File: MAUI/DataGrid/column-drag-and-drop.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+title: Column Drag and Drop in .NET MAUI DataGrid control | Syncfusion
+description: Learn all about column drag and drop support in Syncfusion MAUI DataGrid (SfDataGrid) control and more here.
+platform: MAUI
+control: SfDataGrid
+documentation: ug
+---
+
+# Column Drag and Drop in MAUI DataGrid (SfDataGrid)
+
+The SfDataGrid allows dragging and dropping a column header by setting the `SfDataGrid.AllowDraggingColumn` property to `true.` The drag view is displayed while dragging a column header. Based on the requirements, drag and drop operations can be handled by using the `SfDataGrid.QueryColumnDragging` event.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/column-drag-and-drop.md:12]: The `SfDataGrid` allows you to drag and drop a column header by setting the `SfDataGrid.AllowDraggingColumn` property to true. A drag view is displayed while dragging a column header. You can handle drag and drop operations based on your requirements by using the `SfDataGrid.QueryColumnDragging` event

File: MAUI/DataGrid/column-drag-and-drop.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+{% highlight c# %}
+ dataGrid.AllowDraggingColumn = true;
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="DataGrid column drag and drop" src="Images\column-drag-and-drop\maui-datagrid-column-drag-and-drop.gif" width="404"/>
+
+## Column drag and drop events
+
+The `QueryColumnDragging` event is fired continuously while dragging a column and ends when the dragging ends. By handling the `SfDataGrid.QueryColumnDragging` event, the dragging of a particular column header can be canceled.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/column-drag-and-drop.md:30]: The `QueryColumnDragging` event is fired continuously while dragging and dropping a column. By handling this event, you can cancel the dragging of a particular column header.

File: MAUI/DataGrid/column-drag-and-drop.md [hunk 4]
```diff
@@ -0,0 +1,226 @@
...
+- `From`: Returns the index of the currently dragging column.
+- `To`: Returns the index where you are trying to drop the column.
+- `DraggingAction`: Returns the column dragging details as a `DataGridDragAction`.
+- `DraggingPosition`: Returns the position of the drag view during column drag and drop operations.
+- `CanAutoScroll`: Returns whether auto-scrolling should happen when the column drag view reaches the left or right ends of the `SfDataGrid`.
+- `Cancel`: Returns a boolean property to cancel the event.
+
+## Cancel dragging of a particular column
+
+Dragging of a particular column can be canceled using `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/column-drag-and-drop.md:43]: The dragging of a particular column can be canceled using the `DataGridDragAction` argument in the `QueryColumnDragging` event.

File: MAUI/DataGrid/column-drag-and-drop.md [hunk 5]
```diff
@@ -0,0 +1,226 @@
...
+    if (e.From == 1 && e.DraggingAction == DataGridDragAction.DragStarted) 
+    {
+        e.Cancel = true;
+    }
+} 
+{% endhighlight %}
+
+## Cancel dropping when dragging for a particular column
+
+Dropping when dragging a particular column can be canceled using the `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/column-drag-and-drop.md:61]: The dropping of a particular column when dragging can be canceled using the `DataGridDragAction` argument in the `QueryColumnDragging` event.

File: MAUI/DataGrid/column-drag-and-drop.md [hunk 6]
```diff
@@ -0,0 +1,226 @@
...
+    (e.DraggingAction == DataGridDragAction.DragEnded || e.DraggingAction == DataGridDragAction.Dragging))
+    {
+        e.Cancel = true;
+    }
+}
+{% endhighlight %}
+
+## Cancel dropping for a particular column
+
+The dropping of a particular column can be canceled using the `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/column-drag-and-drop.md:80]: The dropping of a particular column can be canceled using the `DataGridDragAction` argument in the `QueryColumnDragging` event.

File: MAUI/DataGrid/column-drag-and-drop.md [hunk 7]
```diff
@@ -0,0 +1,226 @@
...
+    //e.From returns the index of the dragged column.
+    //e.DraggingAction returns the dragging status of the column.
+    if (e.From == 1 && e.DraggingAction == DataGridDragAction.DragEnded)
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+## Cancel dropping at a particular position
+
+Dropping at a particular position can be canceled using the `DataGridDragAction` argument of the `QueryColumnDragging` event handler.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/column-drag-and-drop.md:96]: The dropping at a particular position can be canceled using the `DataGridDragAction` argument in the `QueryColumnDragging` event.

File: MAUI/DataGrid/column-drag-and-drop.md [hunk 8]
```diff
@@ -0,0 +1,226 @@
...
+    //e.From returns the index of the dragged column.
+    //e.DraggingAction returns the dragging status of the column.
+    if ((e.To == 3 || e.To == 5) && e.DraggingAction == DataGridDragAction.DragEnded)
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+## Cancel dropping of a particular column in a position
+
+Dropping of a particular column in a position can be canceled using the `DataGridDragAction` and `DraggingPosition` arguments of the `QueryColumnDragging` event handler.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/column-drag-and-drop.md:112]: The dropping of a particular column in a position can be canceled using the `DataGridDragAction` and `DraggingPosition` arguments of the `QueryColumnDragging` event.

File: MAUI/DataGrid/column-drag-and-drop.md [hunk 9]
```diff
@@ -0,0 +1,226 @@
...
+            <syncfusion:DataGridStyle ColumnDragViewBackgroundColor="GreenYellow"/>
+        </syncfusion:SfDataGrid.DefaultStyle>
+    </syncfusion:SfDataGrid>
+{% endhighlight %}
+
+<img alt="DataGrid column drag view background color" src="Images\column-drag-and-drop\maui-datagrid-column-drag-view-background-color.png" width="450"/>
+
+## Cancel auto scrolling
+
+Horizontal auto-scrolling of the SfDataGrid during column drag and drop can be canceled using the `CanAutoScroll` argument of the `QueryColumnDragging` event handler.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/column-drag-and-drop.md:216]: Auto-scrolling during column dragging can be disabled using the `CanAutoScroll` argument in the `QueryColumnDragging` event.

File: MAUI/DataGrid/column-drag-and-drop.md [hunk 10]
```diff
@@ -0,0 +1,226 @@
...
+    if (e.DraggingPosition == new Point(100, 100) && e.DraggingAction == DataGridDragAction.DragEnded)
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+## Cancel drag and drop between frozen and non-frozen columns
+
+### Cancel dragging between frozen and non-frozen columns
+
+Dragging between frozen and non-frozen columns can be canceled using the `DataGridDragAction` and `From` arguments of the `QueryColumnDragging` event handler, by checking whether the value of the `From` argument is a frozen column index.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/column-drag-and-drop.md:130]: The dragging between frozen and non-frozen columns can be canceled using the `DataGridDragAction` and `From` arguments of the `QueryColumnDragging` event, by checking whether the value of the `From` argument is a frozen column index.

File: MAUI/DataGrid/column-drag-and-drop.md [hunk 11]
```diff
@@ -0,0 +1,226 @@
...
+    //e.From returns the index of the dragged column.
+    //e.To returns the index of the current column.
+    if (e.From < 2 && e.DraggingAction == DataGridDragAction.DragStarted)
+        e.Cancel = true;
+}
+{% endhighlight %}
+
+### Cancel dropping between frozen and non-frozen columns
+
+Dropping between frozen and non-frozen columns can be canceled using the `DataGridDragAction` and `From` arguments of the `QueryColumnDragging` event handler by checking whether the `e.From` value is a frozen column index.
```
Comment (ashok-kuvaraja) [inline MAUI/DataGrid/column-drag-and-drop.md:148]: The dropping between frozen and non-frozen columns can be canceled using the `DataGridDragAction` and `From` arguments of the `QueryColumnDragging` event by checking whether the `e.From` value is a frozen column index.

**Review by ashok-kuvaraja — CHANGES_REQUESTED** [review]

File: maui-toc.html (hunk 1 above)
Comment (Tamilarasan-Paranthaman) [reply maui-toc.html:495]: Reordered

File: MAUI/DataGrid/column-drag-and-drop.md (hunk 2 above)
Comment (Tamilarasan-Paranthaman) [reply MAUI/DataGrid/column-drag-and-drop.md:12]: Modified

File: MAUI/DataGrid/column-drag-and-drop.md (hunk 3 above)
Comment (Tamilarasan-Paranthaman) [reply MAUI/DataGrid/column-drag-and-drop.md:30]: Modified

File: MAUI/DataGrid/column-drag-and-drop.md (hunk 4 above)
Comment (Tamilarasan-Paranthaman) [reply MAUI/DataGrid/column-drag-and-drop.md:43]: Modified

File: MAUI/DataGrid/column-drag-and-drop.md (hunk 5 above)
Comment (Tamilarasan-Paranthaman) [reply MAUI/DataGrid/column-drag-and-drop.md:61]: Modified

File: MAUI/DataGrid/column-drag-and-drop.md (hunk 6 above)
Comment (Tamilarasan-Paranthaman) [reply MAUI/DataGrid/column-drag-and-drop.md:80]: Modified

File: MAUI/DataGrid/column-drag-and-drop.md (hunk 7 above)
Comment (Tamilarasan-Paranthaman) [reply MAUI/DataGrid/column-drag-and-drop.md:96]: Modified

File: MAUI/DataGrid/column-drag-and-drop.md (hunk 9 above)
Comment (Tamilarasan-Paranthaman) [reply MAUI/DataGrid/column-drag-and-drop.md:216]: Modified

File: MAUI/DataGrid/column-drag-and-drop.md (hunk 10 above)
Comment (Tamilarasan-Paranthaman) [reply MAUI/DataGrid/column-drag-and-drop.md:130]: Modified

File: MAUI/DataGrid/column-drag-and-drop.md (hunk 11 above)
Comment (Tamilarasan-Paranthaman) [reply MAUI/DataGrid/column-drag-and-drop.md:148]: Modified

**Review by ashok-kuvaraja — APPROVED** [review]
The changes are fine.

**Pull request merged**